src/test/unit/algorithms/SDRClassifierTest.cpp

```diff
@@ -0,0 +1,169 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+ * Implementation of unit tests for SDRClassifier
+ */
+
+#include <iostream>
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <nupic/algorithms/ClassifierResult.hpp>
+#include <nupic/algorithms/SDRClassifier.hpp>
+#include <nupic/math/StlIo.hpp>
+#include <nupic/types/Types.hpp>
+#include <nupic/utils/Log.hpp>
+
+using namespace std;
+using namespace nupic;
+using namespace nupic::algorithms::cla_classifier;
+using namespace nupic::algorithms::sdr_classifier;
+
+namespace
+{
+
+  TEST(SDRClassifierTest, Basic)
+  {
+    vector<UInt> steps;
+    steps.push_back(1);
+    SDRClassifier c = SDRClassifier(steps, 0.1, 0.1, 0);
+
+    // Create a vector of input bit indices
+    vector<UInt> input1;
+    input1.push_back(1);
+    input1.push_back(5);
+    input1.push_back(9);
+    ClassifierResult result1;
+    c.compute(0, input1, 4, 34.7, false, true, true, &result1);
+
+    // Create a vector of input bit indices
+    vector<UInt> input2;
+    input2.push_back(1);
+    input2.push_back(5);
+    input2.push_back(9);
+    ClassifierResult result2;
+    c.compute(1, input2, 4, 34.7, false, true, true, &result2);
+
+    {
+      bool foundMinus1 = false;
+      bool found1 = false;
+      for (auto it = result2.begin();
+           it != result2.end(); ++it)
+      {
+        if (it->first == -1)
+        {
+          // The -1 key is used for the actual values
+          ASSERT_EQ(false, foundMinus1)
+            << "already found key -1 in classifier result";
+          foundMinus1 = true;
+          ASSERT_EQ((long unsigned int)5, it->second->size())
```

Comment:
Is this "long unsigned int" cast necessary? I know I often succeed with `ASSERT_EQ(1, myVector.size());`